pkg/proxy/endpointslicecache.go

```diff
@@ -214,16 +219,15 @@ func formatEndpointsList(endpoints []Endpoint) []string {
 }
 
 // endpointSliceCacheKeys returns cache keys used for a given EndpointSlice.
-func endpointSliceCacheKeys(endpointSlice *discovery.EndpointSlice) (types.NamespacedName, string) {
-	if len(endpointSlice.OwnerReferences) == 0 {
-		klog.Errorf("No owner reference set on endpoint slice: %s", endpointSlice.Name)
-		return types.NamespacedName{}, endpointSlice.Name
-	}
-	if len(endpointSlice.OwnerReferences) > 1 {
-		klog.Errorf("More than 1 owner reference set on endpoint slice: %s", endpointSlice.Name)
+func endpointSliceCacheKeys(endpointSlice *discovery.EndpointSlice) (types.NamespacedName, string, error) {
+	var err error
+	serviceName, ok := endpointSlice.Labels[discovery.LabelServiceName]
```

Comment:
the KEP says "For EndpointSlice instances that are not derived from kubernetes Services, the above label must not be applied."... are endpointslice objects without that label filtered out so they aren't handled by this controller?

Reply:
Only endpoint slices that match a service with a selector are currently processed by the controller (matching the corresponding endpoints controller logic). This behavior could be amended/changed if we get any feedback during alpha.